src/currentprocess.rs

```diff
@@ -256,8 +256,7 @@ impl Default for OSProcess {
 #[cfg(feature = "test")]
 pub struct TestProcess {
     pub process: Process,
-    #[allow(dead_code)] // guard is dropped at the end of the test
-    guard: DefaultGuard,
+    _guard: DefaultGuard, // guard is dropped at the end of the test
```

Comment:
This is essentially https://github.com/rust-lang/rustup/pull/3871#discussion_r1641254390.

Reply:
Yeah -- sorry, forgot to fix that.

Reply:
@djc No worries :)